src/library/cmd_reader.h

```diff
@@ -184,5 +165,166 @@ namespace xlang::cmd
         }
 
         std::map<std::string_view, std::vector<std::string>> m_options;
+
+		template<typename O, typename L>
```

Comment:
#1 coding guideline: be consistent! 😉 

In this case, don't mix tabs and spaces.